src/dict.c

```diff
@@ -377,25 +377,30 @@ int dictRehash(dict *d, int n) {
     return 1;
 }
 
-long long timeInMilliseconds(void) {
+long long timeInMicroseconds(void) {
```

Comment:
I've done similar changes to transition to microseconds in https://github.com/redis/redis/pull/12003 but used monotime library instead. Take a look there, you'll get it for free once that PR lands.